### ISSUE TITLE
Blurry look when upscaled by Windows

### ISSUE THREAD

**Issue by bethouxa** [pyfa-org/Pyfa#1225]
## Bug Report
Pyfa looks blurry when scaled up by the OS (on Windows). Especially visible on high DPI displays.

### Actual behavior:
Pyfa looks like this when upscaled : [image](https://puu.sh/weR3E/6c951fbdf1.png)

### Detailed steps to reproduce:
In the Windows control panel, go to "Display", and increase the value of the "Change the size of text, apps..." slider. 

### Release or development git branch? Please note the release version or commit hash:
Release branch : commit  #461f521

### Operating system and version (eg: Windows 10, OS X 10.9, OS X 10.11, Ubuntu 16.10):
Windows 10

**Comment by blitzmann** [comment]
Yeah... unfortunately pyfa is one of those apps that don't play nicely with scaling. This is a limitation of the underlying toolkit, I believe. There's quite a few apps that still look funky when changing the scaling for the desktop (PyCharm, for example, which is the IDE pyfa is developed with, has blurriness), and pyfa is one of them. There may be a few things that can be done, but a quick Google search returns limited information. Further research into this is required. :)

**Comment by Ebag333** [comment]
Unfortunately wxPython was never designed with DPI scaling in mind.  DPI scaling just increases the ratio for the app, so if your app is setup so that it has the best resolution at a 1:1 ratio, then it'll look terrible at a 1:2 ratio.

Modern apps will use extremely high resolution fonts and images even at low resolutions, so you can scale it without losing any quality.

Not much to be done there, since we don't control the font or images.

If your issue is the same as Bart in #1183 then might be able to work around the DPI issue......

**Comment by VioletGiraffe** [comment]
> Modern apps will use extremely high resolution fonts and images even at low resolutions
I don't think that's an accurate nor true description. The fonts, for example, are rendered to the required size. A GUI framework that is DPI-aware will just know that the required size is x1.75 for 175% scaling, it will render a properly sized bitmap that can be mapped to the screen pixel-perfect. A framework that's non-DPI-aware will render at x1 and then the OS will scale up this bitmap, producing blurriness.

**Comment by blitzmann** [comment]
wxWidgets (supposedly) has better support for hidpi in windows in 3.1. Unfortunaltey, wxPython is built with 3.0. I've been pushing Robin here and there in getting a 3.1 version built so that we can try it out. HiDPI is already available in OS x and linux builds, but linux support iirc is still iffy.

**Comment by VioletGiraffe** [comment]
@blitzmann: I see, didn't catch that bit, sorry for nagging, then. And thanks for paying attention to this issue. The program is perfectly usable, of course, but smaller text does get hard to read due to not being as crisp as it's supposed to be.

**Comment by blitzmann** [comment]
No worries at all! It's a question that pops up every now and then, and it's something > definitely want to get fixed at some point once wxPython gets updated

**Comment by DarkFenX** [comment]
Some extra info is in #1957 which i closed

**Comment by DarkFenX** [comment]
Looks like wxpython 4.1.x/4.2.x may fix this:
https://discuss.wxpython.org/t/high-dpi-windows-and-wxwidgets-3-1-x/33061

Not just for windows, for linux too (it has lowres bitmaps, the rest looks alright already): https://github.com/wxWidgets/Phoenix/issues/1327

**Comment by mkaito** [comment]
I wanted to chime in to say that pyfa is barely usable on Linux on my 4k panels. I've resorted to running the (blurry) windows version in a Windows VM for the time being.

It doesn't seem to care what my Xorg DPI is set to.

**Comment by DarkFenX** [comment]
I am using pyfa on my Debian + Gnome + Xorg on 3840x2160 17" display. In display settings I have 200% scaling set, pyfa is properly scaled up. (200% because xorg doesn't support non-integer scaling, and for Gnome via Wayland non-integer scaling has been added as an experimental feature only in [the latest release](https://help.gnome.org/misc/release-notes/3.32/).)

That said, as mentioned in https://github.com/wxWidgets/Phoenix/issues/1327, full hiDPI support in not in wxpython 4.0.6 yet.

Please create separate issue, mentioning screen resolution, distribution you are using, display server (xorg or wayland), dpi/scaling factor, as well as wxpython version.

**Comment by mkaito** [comment]
Actually, the fact that it works for you in Gnome made me dig a little into what Gnome does under the hood. Setting server DPI doesn't work, but setting `GDK_SCALE` and `GDP_DPI_SCALE` does. It's not even blurry like on Windows.

For reference, I'm now running pyfa with `GDK_SCALE=2 GDK_DPI_SCALE=0.5 pyfa`. The first scales the UI elements up, while the second scales the text back a little, which seems to work out fine for my screens.

**Comment by DarkFenX** [comment]
FYI i tried wxpython 4.0.7 under windows - created `hidpi` branch, where I set DPI awareness flag for Windows in 10c12e31962a00940610f5c5bf814e8cbad49c9a (other platforms do not need it).

However, it turns out that `wx.GetApp().GetTopWindow().GetContentScaleFactor()` still returns 1.0 despite actual scaling being 2.0, so we will have to wait until wxpython 4.1.x release, where it will most likely be fixed.

**Comment by DarkFenX** [comment]
4.1.0 was released yesterday. `wx.GetApp().GetTopWindow().GetContentScaleFactor()` returns 2.0 on windows for me, so i guess we could start doing some interface work

**Comment by DarkFenX** [comment]
I started some work in `hidpi` branch, and it worked here and there, but I am blocked by some issues: https://github.com/wxWidgets/Phoenix/issues/1615 + issues listed in #2179
